scala/private/rules/scala_doc.bzl

```diff
@@ -85,19 +103,20 @@ def _scala_doc_impl(ctx):
 
     return [DefaultInfo(files = depset(direct = [output_path]))]
 
-scala_doc = rule(
-    attrs = {
-        "deps": attr.label_list(
-            aspects = [_scaladoc_aspect],
-            providers = [JavaInfo],
-        ),
-        "scalacopts": attr.string_list(),
-        "_scaladoc": attr.label(
-            cfg = "host",
-            executable = True,
-            default = Label("//src/scala/io/bazel/rules_scala/scaladoc_support:scaladoc_generator"),
-        ),
-    },
-    doc = "Generate Scaladoc HTML documentation for source files in from the given dependencies.",
-    implementation = _scala_doc_impl,
-)
+def make_scala_doc_rule(aspect = _scaladoc_transitive_aspect):
+    return rule(
+        attrs = {
+            "deps": attr.label_list(
+                aspects = [aspect],
+                providers = [JavaInfo],
+            ),
+            "scalacopts": attr.string_list(),
+            "_scaladoc": attr.label(
+                cfg = "host",
```

Comment:
nit: `host` seems to be deprecated, and `exec` should be used instead. As you touch this code, I'd appreciate if you can also update it.